ui/lib/pki/addon/routes/application.js

```diff
@@ -0,0 +1,13 @@
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default class PkiRoute extends Route {
+  @service pathHelp;
+  @service secretMountPath;
+
+  beforeModel() {
+    // Must call this promise before the model hook otherwise the model doesn't hydrate from OpenAPI correctly.
+    // only needs to be called once to add the openAPI attributes to the model prototype
+    return this.pathHelp.getNewModel('pki/role', this.secretMountPath.currentPath);
```

Comment:
Good job on finding a work around for ember/openAPI weirdness.

Reply:
Do we only call for role here because of the race conditions? Wondering about the other models that leverage openApi 🤔

Reply:
That's a good question and I think it would be a good idea to use `getNewModel` for any pki model that has `useOpenAPI` returning `true` to be safe. Also this will ensure that it will only register the new models once when the engine route is hit. Sorry for the delayed response :disappointed: